### PR TITLE
Add metric for arguments in training

### PR DESCRIPTION
**Pull request by khai-meetkai**
+ add computing the accuracy of predicting the first token in all arguments in training steps and evaluation steps
+ fix only compute training metrics in training loop, not evaluation loop

### CODE REVIEW

File: functionary/train/train_utils.py
Comment (musab-mk) [inline functionary/train/train_utils.py:1]: Let's put metrics related stuff into metrics.py

**Review by musab-mk — CHANGES_REQUESTED** [review]

**Review by musab-mk — APPROVED** [review]
